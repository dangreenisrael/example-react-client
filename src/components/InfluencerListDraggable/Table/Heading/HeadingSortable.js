import React, { Component } from 'react';
import { DragSource, DropTarget } from 'react-dnd';

const cardSource = {
  beginDrag(props) {
    props.beginDrag(props.id);
    return {
      id: props.id,
      index: props.index
    };
  },
  endDrag(props){
    props.endDrag();
  }
};

const cardTarget = {
  hover(props, monitor) {
    const dragIndex = monitor.getItem().index;
    const hoverIndex = props.index;
    // Don't replace items with themselves
    if (dragIndex === hoverIndex) {
      return;
    }
    // Time to actually perform the action
    props.reorderHeadings(dragIndex, hoverIndex);
    // Note: we're mutating the monitor item here!
    // Generally it's better to avoid mutations,
    // but it's good here for the sake of performance
    // to avoid expensive index searches.
    monitor.getItem().index = hoverIndex;
  },
};

@DropTarget('tableHeading', cardTarget, (connect, monitor)=> ({
  connectDropTarget: connect.dropTarget()
}))
@DragSource('tableHeading', cardSource, (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  isDragging: monitor.isDragging(),
  connectDragPreview: connect.dragPreview(),
}))
export default class Card extends Component {
  render() {
    const blankGif = new Image();
    blankGif.src = "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7";
    this.props.connectDragPreview(blankGif);
    const { text, activeColumn, connectDragSource, connectDropTarget, activeStyle } = this.props;
    const style = (text === activeColumn) ? activeStyle: {cursor:"move"};
    return connectDragSource(connectDropTarget(
        <td style={style}>
          {text}
        </td>,
    ), {dropEffect: "copy"});
  }
}