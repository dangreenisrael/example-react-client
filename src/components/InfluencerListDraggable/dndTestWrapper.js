import TestBackend from 'react-dnd-test-backend';
import { DragDropContext } from 'react-dnd';
import React, {Component} from 'react';
import { Provider } from 'react-redux';
import store from '../../redux/store';

export default (InnerComponent) => {
  @DragDropContext(TestBackend)
  class TestContextContainer extends Component {
    render() {
      return <Provider store={store}>{InnerComponent}</Provider>;
    }
  }
  return TestContextContainer;
}