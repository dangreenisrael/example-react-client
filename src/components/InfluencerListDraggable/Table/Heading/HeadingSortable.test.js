import React from 'react';
import renderer from 'react-test-renderer';
import HeadingSortable from './HeadingSortable';
import dndTestWrapper from '../../dndTestWrapper';

test('HeadingSortable Renders Correctly', () => {
  const HeadingSortableWrapper = dndTestWrapper(<HeadingSortable text="test heading"/>);
  const tree = renderer.create( <HeadingSortableWrapper/>).toJSON();
  expect(tree).toMatchSnapshot();
});