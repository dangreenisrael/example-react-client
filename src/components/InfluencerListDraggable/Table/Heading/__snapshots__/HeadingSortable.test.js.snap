exports[`test HeadingSortable Renders Correctly 1`] = `
<td
  style={
    Object {
      "cursor": "move",
    }
  }>
  test heading
</td>
`;
